File: lowest-missing-pos.java
import java.util.Arrays;

public class MyClass {
    public static void main(String args[])
    {
        int[] input = {-1,0,-3,1,1,4,2};
        Arrays.sort(input);
        
        int length = input.length;
        
        if (input[length - 1] <= 0) { print("no pos"); }
        
        for (int i = 0; i < length - 1; i++)
        {
            int x = input[i];
            int y = input[i+1];
            
            if ((y - x > 1) && !(y + x <= y))
            {
                print(input[i]+1);
            }
        }
        
        print(input[length - 1] + 1);
    }
    
    public static void print(String text)
    {
        System.out.println(text);
        System.exit(0);
    }
    
    public static void print(int num)
    {
        System.out.println(Integer.toString(num));
        System.exit(0);
    }
}

File: lowest-missing-pos.rs
//rust 1.30.0

fn main()
{
    let mut input = [-1,0,-3,1,1,4,2];
    input.sort();
    
    search(&input);
}

fn search(input: &[i32])
{
    let length = input.len();
    
    if input[length - 1] <= 0
    {
        print!("no pos");
        return ();
    }

    for i in 0..(length - 1)
    {
        let x = input[i];
        let y = input[i+1];

        if (y - x > 1) && !(y + x <= y)
        {
            print!("{}", input[i]+1);
            return ();
        }
    }

    print!("{}", input[length - 1] + 1);
    return ();
}